x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/filter/SecurityActionFilter.java

```diff
@@ -110,8 +114,8 @@ public SecurityActionFilter(AuthenticationService authcService, AuthorizationSer
                 listener.onFailure(e);
             }
         } else if (SECURITY_ACTION_MATCHER.test(action)) {
-            if (licenseState.isSecurityEnabled() == false) {
-                listener.onFailure(new ElasticsearchException("Security must be explicitly enabled when using a [" +
+            if (XPackSettings.SECURITY_ENABLED.get(settings) == false) {
+                listener.onFailure(new ElasticsearchException("Security must be enabled when using a [" +
```

Comment:
We could probably skip mentioning the license here since this statement is true for all licenses

Reply:
I think all of this can go.
This filter isn't installed if security is explicitly disabled, so there shouldn't be any way to get into these clauses.

Reply:
Good point! I've changed the methods in this ActionFilter to not check if security is enabled or disabled, assuming it is since the filter _is_ installed